cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -1527,4 +1534,30 @@ private void emitTimeMetrics(String namespace, String appName, String metricName
     long timeTaken = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime);
     metricsCollectionService.getContext(tags).gauge(metricName, timeTaken);
   }
+
+  /**
+   * Marks a given list of application versions as latest.
+   *
+   * @param appIds the list of {@link ApplicationId}s that need to be marked latest
+   * @throws MultipleLatestVersionsException when the list contains multiple versions of the same application
+   * @throws RuntimeException when the update operation fails for any reason.
+   */
+  public void markAppsAsLatest(List<ApplicationId> appIds)
+      throws MultipleLatestVersionsException, IOException, ConflictException {
+    // Validate the appIds do not contain duplicates, i.e. we are not trying to mark
+    // multiple versions of the same application as latest
+    Set<String> validated = new HashSet<>();
```

Comment:
commented earlier how this is incorrect with a suggestion on how to fix. Please don't just mark comments as resolved without resolving them.

Reply:
Sorry, this was a miss. I have updated the method to take namespaceId as a param. The contains check was removed earlier.

Reply:
This has been updated in the latest commit.